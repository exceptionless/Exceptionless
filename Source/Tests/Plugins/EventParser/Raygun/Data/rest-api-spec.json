{
	"occurredOn": "2015-09-08T01:55:28Z",
	"details": {
		"machineName": "ServerMachine1",
		"groupingKey": "ErrorGroup",
		"version": "1.0.0.1",
		"client": {
			"name": "Example Raygun Client",
			"version": "0.0.0.1",
			"clientUrl": "https://raygun.io/docs/integrations/api"
		},
		"error": {
			"innerError": {},
			"data": {
				"example": 5
			},
			"className": "ErrorClass",
			"message": "An error occurred",
			"stackTrace": [
                {
                    "lineNumber": 55,
                    "className": "BrokenService",
                    "columnNumber": 23,
                    "fileName": "BrokenService.cs",
                    "methodName": "BreakEverything<string, int>(Dictionary<string, int> source)"
                }]
		},
        "environment": {
            "processorCount": 4,
            "osVersion": "Windows 10",
            "windowBoundsWidth": 2560,
            "windowBoundsHeight": 1440,
            "browser-Width": 2560,
            "browser-Height": 1440,
            "screen-Width": 2560,
            "screen-Height": 1440,
            "resolutionScale": 1.0,
            "color-Depth": 24,
            "currentOrientation": "Landscape",
            "cpu": "Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz",
            "packageVersion": "package version",
            "architecture": "ARMv7-A",
            "deviceManufacturer": "Nokia",
            "model": "Lumia 920",
            "totalPhysicalMemory": 1024,
            "availablePhysicalMemory": 1024,
            "totalVirtualMemory": 1024,
            "availableVirtualMemory": 1024,
            "diskSpaceFree": [ 50000.52, 2000.104 ],
            "deviceName": "Nexus 7",
            "locale": "en-nz",
            "utcOffset": -12,
            "browser": "Mozilla",
            "browserName": "Netscape",
            "browser-Version": "5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.65 Safari/537.36",
            "platform": "Win32"
        },
		"tags": ["tag1", "tag 2", "tag-3"],
		"userCustomData": {
			"domain": "WORKPLACE",
			"area": "51"
		},
		"request": {
			"hostName": "https://raygun.io",
			"url": "/docs/integrations/api",
			"httpMethod": "POST",
			"iPAddress": "127.0.0.1",
			"queryString": {
				"q": "searchParams"
			},
			"form": {
				"firstName": "Example",
				"lastName": "Person",
				"newsletter": true
			},
			"headers": {
				"Referer": "www.google.com",
				"Host": "raygun.io"
			},
			"rawData": "{\"Test\": 5}"
		},
		"response": {
			"statusCode": 500 
		},
		"user": {
			"identifier": "123456789",
			"isAnonymous": false,
			"email": "test@example.com",
			"fullName": "Test User",
			"firstName": "Test",
			"uuid": "783491e1-d4a9-46bc-9fde-9b1dd9ef6c6e"
		}
	}
}