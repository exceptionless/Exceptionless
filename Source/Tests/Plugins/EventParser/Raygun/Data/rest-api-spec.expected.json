{
  "is_first_occurrence": false,
  "is_fixed": false,
  "is_hidden": false,
  "created_utc": "0001-01-01T00:00:00",
  "type": "error",
  "date": "2015-09-08T01:55:28+00:00",
  "tags": [
    "tag1",
    "tag 2",
    "tag-3"
  ],
  "message": "An error occurred",
  "data": {
    "@stack": "ErrorGroup",
    "domain": "WORKPLACE",
    "area": "51",
    "Client": {
      "name": "Example Raygun Client",
      "version": "0.0.0.1",
      "client_url": "https://raygun.io/docs/integrations/api"
    },
    "Response": {
      "status_code": 500
    },
    "@error": {
      "message": "An error occurred",
      "type": "ErrorClass",
      "data": {
        "example": 5
      },
      "stack_trace": [
        {
          "file_name": "BrokenService.cs",
          "line_number": 55,
          "column": 23,
          "is_signature_target": false,
          "declaring_type": "BrokenService",
          "name": "BreakEverything<string, int>(Dictionary<string, int> source)",
          "module_id": -1
        }
      ],
      "target_method": {
        "is_signature_target": false,
        "declaring_type": "BrokenService",
        "name": "BreakEverything<string, int>(Dictionary<string, int> source)",
        "module_id": -1
      }
    },
    "@environment": {
      "processor_count": 4,
      "total_physical_memory": 1073741824,
      "available_physical_memory": 1073741824,
      "process_memory_size": 0,
      "architecture": "ARMv7-A",
      "o_s_name": "Win32",
      "o_s_version": "Windows 10",
      "machine_name": "ServerMachine1",
      "install_id": "783491e1-d4a9-46bc-9fde-9b1dd9ef6c6e",
      "data": {
        "AvailableVirtualMemory": 1073741824,
        "Browser": "Mozilla",
        "BrowserName": "Netscape",
        "BrowserVersion": "5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.65 Safari/537.36",
        "BrowserHeight": 1440,
        "BrowserWidth": 2560,
        "ColorDepth": 24,
        "Cpu": "Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz",
        "CurrentOrientation": "Landscape",
        "DeviceManufacturer": "Nokia",
        "DeviceName": "Nexus 7",
        "DiskSpaceFree": [
          50000.52,
          2000.104
        ],
        "Locale": "en-nz",
        "Model": "Lumia 920",
        "PackageVersion": "package version",
        "ResolutionScale": 1.0,
        "ScreenHeight": 1440,
        "ScreenWidth": 2560,
        "TotalVirtualMemory": 1073741824,
        "WindowBoundsHeight": 1440.0,
        "WindowBoundsWidth": 2560.0
      }
    },
    "@request": {
      "user_agent": "5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.65 Safari/537.36",
      "http_method": "POST",
      "is_secure": true,
      "host": "https://raygun.io",
      "port": 443,
      "path": "/docs/integrations/api",
      "referrer": "www.google.com",
      "client_ip_address": "127.0.0.1",
      "post_data": {
        "first_name": "Example",
        "last_name": "Person",
        "newsletter": true
      },
      "query_string": {
        "q": "searchParams"
      },
      "data": {
        "Headers": {
          "referer": "www.google.com",
          "host": "raygun.io"
        },
        "RawData": "{\"Test\": 5}",
        "@browser": "Mozilla",
        "@browser_version": "5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.65 Safari/537.36",
        "BrowserHeight": 1440,
        "BrowserWidth": 2560,
        "@device": "Nokia Nexus 7"
      }
    },
    "@user": {
      "identity": "test@example.com",
      "name": "Test User",
      "data": {
        "Email": "test@example.com",
        "FirstName": "Test",
        "FullName": "Test User",
        "Identifier": "123456789",
        "IsAnonymous": false,
        "Uuid": "783491e1-d4a9-46bc-9fde-9b1dd9ef6c6e"
      }
    },
    "@version": "1.0.0.1"
  }
}